app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt

```diff
@@ -415,26 +415,24 @@ class ConversationListViewModel @Inject constructor(
     fun moveConversationToFolder(id: String = "") {
     }
 
-    fun moveConversationToArchive(
-        conversationId: ConversationId,
-        isArchiving: Boolean,
-        timestamp: Long = DateTimeUtil.currentInstant().toEpochMilliseconds()
-    ) {
-        viewModelScope.launch {
-            requestInProgress = true
-            val result = withContext(dispatcher.io()) { updateConversationArchivedStatus(conversationId, isArchiving, timestamp) }
-            requestInProgress = false
-            when (result) {
-                is ArchiveStatusUpdateResult.Failure -> {
-                    homeSnackBarState.emit(HomeSnackbarState.UpdateArchivingStatusError(isArchiving))
-                }
+    fun moveConversationToArchive(dialogState: DialogState, timestamp: Long = DateTimeUtil.currentInstant().toEpochMilliseconds()) =
+        with(dialogState) {
+            viewModelScope.launch {
+                val isArchiving = !isArchived
+                requestInProgress = true
+                val result = withContext(dispatcher.io()) { updateConversationArchivedStatus(conversationId, isArchiving, timestamp) }
```

Comment:
I think it's main safe, we don't need to switch dispatcher here as it's already done in lower levels(Dao classes)

Reply:
You are right. I followed a bit the pattern for `clearConversationContent` method in case there was something else I was missing. Should I also remove there that context switching as I guess it is also done in DB?

Reply:
Done removing them and it seems everything works like normal. Yay 🎉